### PR TITLE
docs - updates to Encrypting Client/Server Connections

### PR DESCRIPTION
**Pull request by mkiyama**
clarified/enhanced SSL configuration information.

This will be backported to 6X_STABLE and 5X_STABLE

Link to the HTML output on a temporary GPDB doc review site.
https://docs-msk-gpdb6-dev.cfapps.io/7-0/admin_guide/client_auth.html#topic5

### CODE REVIEW

File: gpdb-doc/dita/admin_guide/client_auth.xml [hunk 1]
```diff
@@ -302,20 +301,37 @@ max_prepared_transactions=100
...
-        certificate authority (CA) should be used in production, so the client can verify the
-        identity of the server. Either a global or local CA can be used. If all the clients are
-        local to the organization, a local CA is recommended.</p>
+      <p>Enabling Greenplum Database in SSL mode requires the following items.<ul
+          id="ul_ocx_df1_dlb">
+          <li>OpenSSL installed on both the client and the master server hosts (master and standby
+            master).</li>
+          <li>The SSL files <filepath>server.key</filepath> (server private key) and
+              <filepath>server.crt</filepath> (server certificate) should be correctly generated for
+            the master host and standby master host. <ul id="ul_pwr_dg1_dlb">
```
Comment (lenapivotal) [inline gpdb-doc/dita/admin_guide/client_auth.xml]: Any thoughts on how to verify the  "correctly generated" files?

File: gpdb-doc/dita/admin_guide/client_auth.xml [hunk 2]
```diff
@@ -302,20 +301,37 @@ max_prepared_transactions=100
...
+              <li>The private key should not be protected with a passphrase. The server does not
+                prompt for a passphrase for the private key, and Greenplum Database start up fails
+                with an error if one is required.</li>
+              <li>On a production system, there should be a key and certificate pair for the master
+                host and a pair for the standby master host with a subject CN (Common Name) for the
+                master host and standby master host.</li>
```
Comment (lenapivotal) [inline gpdb-doc/dita/admin_guide/client_auth.xml]: I'm not clear if the requirement is the same for the master host and standby host, is it?

File: gpdb-doc/dita/admin_guide/client_auth.xml [hunk 3]
```diff
@@ -302,20 +301,37 @@ max_prepared_transactions=100
...
+              identity of the server. Either a global or local CA can be used. If all the clients
+              are local to the organization, a local CA is recommended.</p></li>
+          <li>The <filepath>server.key</filepath> and <filepath>server.crt</filepath> have been
+            added to the master data directory on both the master host and standby master host. When
+            starting in SSL mode, the Greenplum Database master looks for
+              <filepath>server.key</filepath> and <filepath>server.crt</filepath>. Greenplum
+            Database does not start if the files are not in the master data directory of both the
+            master host and standby master host. <p>Also, if you use other SSL authentication files
+              such as <codeph>root.crt</codeph> (trusted certificate authorities), the files must be
+              on both the master host and standby master host.</p></li>
```
Comment (lenapivotal) [inline gpdb-doc/dita/admin_guide/client_auth.xml]: We could move this after the first bullet, as it's related to the same files and it feels similar context?

File: gpdb-doc/dita/admin_guide/client_auth.xml [hunk 4]
```diff
@@ -302,20 +301,37 @@ max_prepared_transactions=100
...
+        </ul> Greenplum Database can be started with SSL enabled by setting the server configuration
+        parameter <codeph>ssl=on</codeph> in the <codeph>postgresql.conf</codeph> on the master and
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/client_auth.xml]: postgresql.conf -> postgresql.conf file

File: gpdb-doc/dita/admin_guide/client_auth.xml (hunk 1 above)
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/client_auth.xml]: The two sub-bullets describe the GPDB items
--no passphrase
--CN uses the host names for the master and standby

Comment (lenapivotal) [reply gpdb-doc/dita/admin_guide/client_auth.xml]: You are right, I missed the formatting when reading in GitHub... Apologies.

File: gpdb-doc/dita/admin_guide/client_auth.xml (hunk 3 above)
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/client_auth.xml]: This was my thinking on the current order
--host OS configuration
--generate SSL .key and .crt
--adding .key and .crt to host systems

Comment (lenapivotal) [reply gpdb-doc/dita/admin_guide/client_auth.xml]: Totally, I should have read it on the staging app, looks good.

File: gpdb-doc/dita/admin_guide/client_auth.xml (hunk 2 above)
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/client_auth.xml]: The master and standby master should have their own set of .crt and .key files. The files should be created using the host name of the master and standby master.

**Comment by mkiyama** [comment]
@ysung-pivotal Could you please review this update
and let me know what changes re requried?

thanks

**Pull request merged**